tldw_Server_API/tests/Watchlists/test_admin_runs_roles_normalization.py

```
@@ -0,0 +1,42 @@
+    """
+    p = Path("tldw-frontend/pages/admin/watchlists-runs.tsx")
+    assert p.exists(), "watchlists-runs.tsx not found"
+    text = p.read_text(encoding="utf-8")
+
+    # Two acceptable implementations:
+    # 1) Local normalization in page (legacy): look for _rawRoles normalization
+    # 2) Centralized hook (preferred): useIsAdmin imported/used
+    if "useIsAdmin" in text:
+        # Ensure the hook is imported and used
+        assert "useIsAdmin" in text
+        assert "const userIsAdmin" in text and "useIsAdmin(" in text or "useIsAdmin()" in text
```

Ruff (0.14.2)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)